yroots/IntervalChecks.py

```diff
@@ -545,373 +545,356 @@ def quadratic_check_3D(test_coeff, intervals, change_sign, tol):
                 if min_satisfied and max_satisfied:
-                    mask.append(True)
                     continue
 
         #Add the z constant boundaries
         #The partials with respect to x and y are zero
         #Dx: 4c7x +  c4y  = -c1 - c5z    Matrix inverse is [4c8  -c4]
         #Dy:  c4x + 4c8y  = -c2 - c6z                      [-c4  4c7]
-        det = 16*c7*c8-c4**2
-        if det != 0:
-            z = interval[0][2]
-            x = (-4*c8*(c1+c5*z) +   c4*(c2+c6*z))/det
-            y = (c4*(c1+c5*z)    - 4*c7*(c2+c6*z))/det
-            if interval[0][0] < x < interval[1][0] and interval[0][1] < y < interval[1][1]:
-                eval = eval_func(x,y,z)
+        if fix_z_det != 0:
+            c1_c5z0 = c[1]+c[5]*z0
+            c2_c6z0 = c[2]+c[6]*z0
+            x = (-kk8*c1_c5z0 +   c[4]*c2_c6z0)/fix_z_det
+            y = (c[4]*c1_c5z0 -    kk7*c2_c6z0)/fix_z_det
+            if x0 < x < x1 and y0 < y < y1:
+                eval = eval_func(x,y,z0)
                 min_satisfied = min_satisfied or eval < other_sum
                 max_satisfied = max_satisfied or eval > -other_sum
                 if min_satisfied and max_satisfied:
-                    mask.append(True)
                     continue
-            z = interval[1][2]
-            x = (-4*c8*(c1+c5*z) +   c4*(c2+c6*z))/det
-            y = (c4*(c1+c5*z)    - 4*c7*(c2+c6*z))/det
-            if interval[0][0] < x < interval[1][0] and interval[0][1] < y < interval[1][1]:
-                eval = eval_func(x,y,z)
+            c1_c5z1 = c[1]+c[5]*z1
+            c2_c6z1 = c[2]+c[6]*z1
+            x = (-kk8*c1_c5z1 +   c[4]*c2_c6z1)/fix_z_det
+            y = (c[4]*c1_c5z1 -    kk7*c2_c6z1)/fix_z_det
+            if x0 < x < x1 and y0 < y < y1:
+                eval = eval_func(x,y,z1)
                 min_satisfied = min_satisfied or eval < other_sum
                 max_satisfied = max_satisfied or eval > -other_sum
                 if min_satisfied and max_satisfied:
-                    mask.append(True)
                     continue
 
         #Add the interior value
-        if int_x is not None and interval[0][0] < int_x < interval[1][0] and interval[0][1] < int_y < interval[1][1] and\
-                interval[0][2] < int_z < interval[1][2]:
+        if x0 < int_x < x1 and y0 < int_y < y1 and\
+                z0 < int_z < z1:
             eval = eval_func(int_x,int_y,int_z)
             min_satisfied = min_satisfied or eval < other_sum
             max_satisfied = max_satisfied or eval > -other_sum
             if min_satisfied and max_satisfied:
-                mask.append(True)
                 continue
 
-        #no root
-        mask.append(False)
+        # No root possible
+        mask[i] = False
 
     return mask
 
+@Memoize
+def get_fixed_vars(dim):
+    """Used in quadratic_check_nd to iterate through the boundaries of the domain.
+
+    Parameters
+    ----------
+    dim
```

Comment:
From a documentation standpoint, I think we technically want this to say something like:
```
dim : int
    The dimension of the domain/system.

```
As it's pretty self-explanatory, I don't think it's a major issue.